### PR TITLE
feat(reat,es6,node,base,deps): Move all warnings to errors

### LINKED ISSUE #11: Can we move to no warnings?
Warnings are universally ignored. Can we decide whether the warnings are important or not and change them to errors?

### PR DESCRIPTION
**Pull request by paulkaplan**
Make all previous warning rules into errors except the todo rule

Fixes #11, breaking change include eslint4.0 and other dep updates, and all rule changes

See https://github.com/LLK/eslint-config-scratch/issues/11#issue-249406530 for discussion about the update. 

@rschamp @cwillisf i'm not familiar with the way this repo handles versioning, so I just followed the instructions from the README. Hopefully I did it right. 🌮  You might want to double check the dependency updates, those were done manually to the most recent releases.

### CODE REVIEW

**Comment by rschamp** [comment]
@paulkaplan You are missing `BREAKING CHANGE: ` in the commit description, which will trigger semantic release to make a new major version.  I believe we can add it in the merge commit message though.

**Comment by rschamp** [comment]
Oh. and I see text "breaking change" is there, but I don't think the parser is smart enough to use that... See https://github.com/bcoe/conventional-changelog-standard/blob/master/convention.md#footer, it's pretty specific about formatting.

File: index.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ module.exports = {
         'no-loop-func': [2],
         'no-multi-spaces': [2],
         'no-multi-str': [2],
-        'no-new': [1],
+        'no-new': [2],
         'no-proto': [2],
         'no-restricted-properties': [1, {
```
Comment (rschamp) [inline index.js]: We should also remove this rule altogether instead of leaving it as a warning.

Comment (paulkaplan) [inline index.js]: Fixed in 4b39b76

**Comment by paulkaplan** [comment]
Fixed commit message with 4b39b76. I'm sad that the CLI for `git cz` doesn't do that automatically :(

**Pull request merged**